app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -172,6 +172,10 @@ struct DeviceDelegateOculusVR::State {
     vrapi_SetPropertyInt(&java, VRAPI_BLOCK_REMOTE_BUTTONS_WHEN_NOT_EMULATING_HMT, 0);
     // Reorient the headset after controller recenter.
     vrapi_SetPropertyInt(&java, VRAPI_REORIENT_HMD_ON_CONTROLLER_RECENTER, 1);
+    // Enable fixed-foveated rendering.
+    // TODO - This should be user configurable. VRAPI_SYS_PROP_FOVEATION_AVAILABLE
```

Comment:
Check `VRAPI_SYS_PROP_FOVEATION_AVAILABLE` before calling `vrapi_SetPropertyInt(&java, VRAPI_FOVEATION_LEVEL, 2);`

Reply:
Initially (my earlier patch), I was attempting to check VRAPI_SYS_PROP_FOVEATION_AVAILABLE before enabling the UI, but it would require many more changes (eg, getting the value asynchronously from the render thread while building the UI).  Setting this value when it is not supported will simply be a NULL operation.  I've hidden the UI when not using the Oculus API, but perhaps we could enhance this at a later time, together with better adaptive UI selections for other rendering features?

Reply:
I imagine the only case right now where VRAPI_SYS_PROP_FOVEATION_AVAILABLE  would return false would be for GearVR / clip-in headsets using the Oculus API.  In those cases with the current patch, the option would appear but not have any effect.